File: suites/03_data_structures/comparisons/linked_list.cpp
#include <cstdio>
static long long pointer_chase(long long nodes, long long rounds) {
    long long cursor = 1, checksum = 0;
    for (long long s = 0; s < rounds; ++s) {
        cursor = ((cursor * 1103515245LL) + 12345) % nodes;
        checksum += cursor;
    }
    return checksum;
}
int main() { printf("%lld\n", pointer_chase(1000003, 8000000)); return 0; }

File: suites/03_data_structures/comparisons/linked_list.rs
fn pointer_chase(nodes: i64, rounds: i64) -> i64 {
    let mut cursor: i64 = 1;
    let mut checksum: i64 = 0;
    for _ in 0..rounds {
        cursor = ((cursor.wrapping_mul(1103515245)) + 12345) % nodes;
        checksum += cursor;
    }
    checksum
}
fn main() { println!("{}", pointer_chase(1000003, 8000000)); }
